### ISSUE TITLE
Set consistency level for Datastax Java driver using kundera

### ISSUE THREAD

**Issue by ansarrafique** [Impetus/kundera#571]
I have just switched to Kundera 2.11 to use Datastax Java driver underneath. Is there anyway to specify Consistency level for Datastax Java driver using Kundera ? For thrift I have used

```
        EntityManagerFactory emf = Persistence.createEntityManagerFactory("cassandra_pu");
        mgr = emf.createEntityManager();
            Map<String, Client> clientMap = (Map<String, Client>) mgr.getDelegate();
            ThriftClient tc = (ThriftClient) clientMap.get("cassandra_pu");
            //tc.setCqlVersion(CassandraConstants.CQL_VERSION_3_0);
            tc.setConsistencyLevel(EMFConfig.consistencyLevel);
```

return mgr;

Also, em.persist() and em.find() are also supported for Datastax Java driver ?

Thanks,

**Comment by mevivs** [comment]
Yes. It will work in same way for em.persist()/em.find() and consistency level as well.

-Vivek

**Comment by ansarrafique** [comment]
Thanks Vivek, but how to specify consistency via client  for the entity manager? Datastax allows to specify consistency for each operation. Do you have simple example ?

Thanks,

**Comment by ansarrafique** [comment]
Does this property works for each driver supported underneath ? e.g Thrift, Pelops, and Datastax Java driver ? 

em.setProperty("consistency.level",  ConsistencyLevel.QUORUM);

**Comment by mevivs** [comment]
Yes

**Comment by ansarrafique** [comment]
Thanks Vivek, Is there any way to check the current consistency level for Kundera entity manager at run time ? I doubt that em.setProperty("consistency.level", ConsistencyLevel.QUORUM) doesn't work properly because it takes the same amount of time to execute queries both using consistency level ONE or QUORUM. 

There was a difference when I set Consistency level by getting the thrift client using entity manager and set it for thrift client instead.

//Map<String, Client> clientMap = (Map<String, Client>) mgr.getDelegate();
ThriftClient tc = (ThriftClient) clientMap.get("cassandra_pu");
tc.setCqlVersion(CassandraConstants.CQL_VERSION_3_0);
tc.setConsistencyLevel(EMFConfig.consistencyLevel);

Thanks,

Thanks,

**Comment by mevivs** [comment]
https://github.com/impetus-opensource/Kundera/blob/trunk/src/kundera-cassandra/cassandra-ds-driver/src/main/java/com/impetus/kundera/client/cassandra/dsdriver/DSClient.java#L521

Not sure about performance, but it is in consistent with Thrift and pelops.

To verify, whether it is being set or not? Try using reflection to invoke getConsistencyLevel()
https://github.com/impetus-opensource/Kundera/blob/trunk/src/kundera-cassandra/cassandra-core/src/main/java/com/impetus/client/cassandra/CassandraClientBase.java#L1164

HTH,
-Vivek

**Comment by chhavigangwal** [comment]
Fixed. Released with Kundera-2.12.
